File: sort/insertion_sort/R/inserTionSort.R
insertionSort <- function(arr) {
  
  for (j in 2:length(arr)) {
    x = arr[j] 
    i = j - 1 
    while (i > 0 && arr[i] >x) {
      arr[(i + 1)] = arr[i]
      i = i - 1 
    }
    arr[(i + 1)] = x
  }
  assign("arr",arr,envir = parent.frame())          
  
}

main <- function() {
  
  prompt <- "Enter numbers to be sorted "
  arr <- as.integer(strsplit(readline(prompt), " ")[[1]])
  insertionSort(arr)
  print(" Array in sorted order : ")
  print(arr)
}

main()

File: sort/insertion_sort/Java/insertionSort.java
public class insertion_sort 
{ 
    void sort(int arr[]) 
    { 
        int n = arr.length; 
        for (int i=1; i<n; ++i) 
        { 
            int item = arr[i]; 
            int j = i-1; 
            while (j>=0 && arr[j] > item) 
            { 
                arr[j+1] = arr[j]; 
                j = j-1; 
            } 
            arr[j+1] = item; 
        } 
    } 

    static void print(int arr[]) 
    { 
        int n = arr.length; 
        for (int i=0; i<n; ++i) 
            System.out.print(arr[i] + " "); 
  
        System.out.println(); 
    } 
    public static void main(String args[]) 
    {         
        int arr[] = {17, 14, 10, 9, 6, 1, 3, 7, 8}; 
  
        insertion_sort object = new insertion_sort();         
        object.sort(arr); 
          
        print(arr); 
    } 
}